Source/WebCore/platform/graphics/coretext/FontCoreText.cpp

```diff
@@ -516,7 +516,7 @@ RefPtr<Font> Font::createFontWithoutSynthesizableFeatures() const
     float size = m_platformData.size();
     CTFontSymbolicTraits fontTraits = CTFontGetSymbolicTraits(getCTFont());
     RetainPtr<CTFontRef> ctFont = createCTFontWithoutSynthesizableFeatures(getCTFont());
-    return createDerivativeFont(ctFont.get(), size, m_platformData.orientation(), fontTraits, m_platformData.syntheticBold(), m_platformData.syntheticOblique(), m_platformData.widthVariant(), m_platformData.textRenderingMode(), m_platformData.customPlatformData());
+    return createDerivativeFont(ctFont.get(), size, m_platformData.orientation(), fontTraits, m_platformData.syntheticBold(), m_platformData.syntheticOblique(), m_platformData.widthVariant(), m_platformData.textRenderingMode(), m_platformData.protectedCustomPlatformData().get());
```

Comment:
The change itself looks good. However, the `FontCustomPlatformData` raw ptr is used by `createDerivativeFont` for instantiating a `FontPlatformData` object which has a ctor [1] that accepts such raw ptr. This ctor calls yet another `FontPlatformData` ctor [2] that takes a `RefPtr`. Shouldn't we change ctor 1 to get a `RefPtr` directly instead of a RawPtr and then move it?

Reply:
Oh, I didn't realize that. Yes, would do that.

Reply:
Hm... this is annoying. We can't quite do that without including FontCustomPlatformData.h in FontPlatformData.h or else we'd get this build error:
```
/Volumes/Data/safari/OpenSource/WebKitBuild/Debug/usr/local/include/wtf/Ref.h:62:16: error: member access into incomplete type 'const WebCore::FontCustomPlatformData'
   62 |             ptr->deref();
      |                ^
In file included from /Volumes/Data/safari/OpenSource/WebKitBuild/Debug/DerivedSources/WebCore/unified-sources/UnifiedSource9-mm.mm:1:
In file included from /Volumes/Data/safari/OpenSource/Source/WebCore/WebCorePrefix.h:165:
In file included from /Volumes/Data/safari/OpenSource/WebKitBuild/Debug/usr/local/include/wtf/HashMap.h:25:
In file included from /Volumes/Data/safari/OpenSource/WebKitBuild/Debug/usr/local/include/wtf/HashTable.h:33:
In file included from /Volumes/Data/safari/OpenSource/WebKitBuild/Debug/usr/local/include/wtf/HashTraits.h:26:
In file included from /Volumes/Data/safari/OpenSource/WebKitBuild/Debug/usr/local/include/wtf/HashFunctions.h:26:
/Volumes/Data/safari/OpenSource/WebKitBuild/Debug/usr/local/include/wtf/RefPtr.h:60:47: note: in instantiation of member function 'WTF::DefaultRefDerefTraits<const WebCore::FontCustomPlatformData>::derefIfNotNull' requested here
   60 |     ALWAYS_INLINE ~RefPtr() { RefDerefTraits::derefIfNotNull(PtrTraits::exchange(m_ptr, nullptr)); }
      |                                               ^
In file included from /Volumes/Data/safari/OpenSource/WebKitBuild/Debug/DerivedSources/WebCore/unified-sources/UnifiedSource9-mm.mm:3:
In file included from /Volumes/Data/safari/OpenSource/Source/WebCore/editing/cocoa/AttributedString.mm:31:
In file included from /Volumes/Data/safari/OpenSource/Source/WebCore/platform/graphics/Font.h:26:
/Volumes/Data/safari/OpenSource/Source/WebCore/platform/graphics/FontPlatformData.h:253:327: note: in instantiation of member function 'WTF::RefPtr<const WebCore::FontCustomPlatformData>::~RefPtr' requested here
  253 |     WEBCORE_EXPORT FontPlatformData(RetainPtr<CTFontRef>&&, float size, bool syntheticBold = false, bool syntheticOblique = false, FontOrientation = FontOrientation::Horizontal, FontWidthVariant = FontWidthVariant::RegularWidth, TextRenderingMode = TextRenderingMode::AutoTextRendering, RefPtr<const FontCustomPlatformData> = nullptr);
      |                                                                                                                                                                                                                                                                                                                                       ^
/Volumes/Data/safari/OpenSource/Source/WebCore/platform/graphics/FontPlatformData.h:81:8: note: forward declaration of 'WebCore::FontCustomPlatformData'
   81 | struct FontCustomPlatformData;
      |        ^
1 error generated.
```

Reply:
or so I thought. But `FontCustomPlatformData.h` includes `FontPlatformData.h` so it's more hairy than that. I don't think there is a conceivable way we could do this short of merging those two header files.

Reply:
Fair enough, I guess the patch itself is fine.